### DISCUSSION TITLE
How to access current locale from the middleware?

### DISCUSSION THREAD

**Discussion by DamianGlowala** [nuxt-modules/i18n#1670] [Q&A]
Placing `const { locale } = useNuxtApp()` within global route middleware does not work. How can I access the locale?

**Comment by ineshbose** [comment]
Hi @DamianGlowala,

You can access the locale using `useI18n()` composable - is that not working in middleware?
An example of its usage can be seen in the example code for `pages/index.vue` (second in the section): https://v8.i18n.nuxtjs.org/getting-started/basic-usage#translate-with-vue-i18n

**Reply by DamianGlowala** [reply]
Hi, thanks for your answer! However, page is something different to a middleware (a file placed in the `middleware/` directory).

**Reply by ineshbose** [reply]
I understand, but I believe it to be usable within middleware files as well. This is how it should look:

```js
export default defineNuxtRouteMiddleware((to, from) => {
  const { locale } = useI18n()
  // rest of your function
})
```

Perhaps if that's still not the case, if you'd be able to share your code and the output you're getting?

**Reply by kazupon** [reply]
You can use `$i18n` that is exported `useNuxtApp`

```ts
export default defineNuxtRouteMiddleware((to, from) => {
  const { $i18n } = useNuxtApp()
  console.log('locale', $i18n.locale)
})
```

**Reply by kazupon** [reply]
We will release v8.0.0-beta.5 soon.
That release will also resolve the completion issue #1621.

**Reply by DamianGlowala** [reply]
@kazupon, is this a known issue? Type hint of `locale` says it's a `string`, but it appears to be a ref instead. `$i18n.locale.value` returns the correct value (e.g. 'en').

![image](https://user-images.githubusercontent.com/48835293/206465465-4c081e60-be10-4eb5-8a05-d28959a99cff.png)

**Reply by kazupon** [reply]
@DamianGlowala 
Hi! Thank you for your feedback as always!

I think that issue is fixed latest edge version.
Please check it!

**Reply by DamianGlowala** [reply]
@kazupon types are correct now, thank you! :)
